core/src/avm2/class.rs

```diff
@@ -78,7 +78,7 @@ pub struct Class<'gc> {
     name: QName<'gc>,
 
     /// The type parameter for this class (only supported for Vector)
-    param: Option<GcCell<'gc, Class<'gc>>>,
+    param: Option<Option<GcCell<'gc, Class<'gc>>>>,
```

Comment:
This is a funny type and (without reading the usages, I admit) I'm struggling to come up with a meaning of the three potential states.

Can we maybe change this to a 3 state enum so we know what `Some(None)` vs `None` actually means?

Reply:
(not blocking PR on this, can fix up later!)